### PR TITLE
Feature/2353 add queue facet

### PR DESCRIPTION
**Pull request by philipkcl**
# Add queue facet to background jobs

*Please don't delete any sections when completing this PR template; instead enter **N/A** for checkboxes or sections which are not applicable, unless otherwise stated below*

See https://github.com/DOAJ/doajPM/issues/2353

Describe the scope/purpose of the PR here in as much detail as you like
* add new field `queue_type` to model `BackgroundJob`
* updated background job search page for new field
* do we need script to assign value of `queue_type` of old background job record?

## Categorisation

This PR...
- [ ] has scripts to run
- [x] has migrations to run
- [ ] adds new infrastructure
- [ ] changes the CI pipeline
- [ ] affects the public site
- [x] affects the editorial area
- [ ] affects the publisher area

## Basic PR Checklist

- [x] FeatureMap annotations have been added
- [x] Unit tests have been added/modified
- [x] **N/A** Functional tests have been added/modified
- [x] Code has been run manually in development, and functional tests followed locally
- [x] No deprecated methods are used
- [x] No magic strings/numbers - all strings are in `constants` or `messages` files
- [x] **N/A** ES queries are wrapped in a Query object rather than inlined in the code
- [x] Where possible our common library functions have been used (e.g. dates manipulated via `dates`)
- [ ] If needed, migration has been created and tested locally
- [x] Release sheet has been created, and completed as far as is possible https://docs.google.com/spreadsheets/d/1Bqx23J1MwXzjrmAygbqlU3YHxN1Wf7zkkRv14eTVLZQ/edit#gid=41046773&range=A1
- [x] **N/A** Documentation updates - if needed - have been identified and prepared for inclusion into main documentation (e.g. added and highlighted/commented as appropriate to this PR)
    - [x] Core model documentation: https://docs.google.com/spreadsheets/d/1lun2S9vwGbyfy3WjIjgXBm05D-3wWDZ4bp8xiIYfImM/edit
    - [x] Events and consumers documentation: https://docs.google.com/spreadsheets/d/1oIeG5vg-blm2MZCE-7YhwulUlSz6TOUeY8jAftdP9JE/edit
- [x] There has been a recent merge up from `develop` (or other base branch)
  - 2022-11-05

## Testing

A quick review of the background jobs search to make sure everything is still working as expected.  Functional test run master file here https://docs.google.com/spreadsheets/d/1iseLQ07IngWRxR7jfYBrMDu4J1cIJl1rZxJ0r79nFV0/edit#gid=0

## Deployment

What deployment considerations are there? (delete any sections you don't need)

### Configuration changes

N/A

### Migrations

**TODO**: @Steven-Eardley is providing a migration to set the appropriate mappings

### Monitoring

N/A

### New Infrastructure

N/A

### Continuous Integration

N/A

### CODE REVIEW

File: portality/tasks/ingestarticles.py [hunk 1]
```diff
@@ -562,7 +558,10 @@ def __fail(record, previous, error):
             return __fail(record, previous, error="please check it before submitting again; " + str(e))
 
 
-@main_queue.task(**configure("ingest_articles"))
+huey_helper = IngestArticlesBackgroundTask.create_huey_helper(main_queue)
+
+
+@huey_helper.register_execute(is_load_config=True)
 @write_required(script=True)
```
Comment (richard-jones) [inline portality/tasks/ingestarticles.py:565]: I can see `register_execute` is also calling `write_required` so can we remove it from here?

**Review by richard-jones — CHANGES_REQUESTED** [review]
I think only one minor change/question about whether the `write_required` decorator is required in some places.  I notice that it has been removed from others, so it's probably just an oversight, and can be removed from the tasks wherever it is used.

File: portality/static/js/edges/admin.background_jobs.edge.js [hunk 2]
```diff
@@ -185,6 +186,20 @@ $.extend(true, doaj, {
...
+                    display: "Queue Type",
+                    deactivateThreshold : 1,
+                    renderer: edges.bs3.newRefiningANDTermSelectorRenderer({
+                        controls: true,
+                        open: false,
+                        togglable: true,
+                        countFormat: countFormat,
+                        hideInactive: true
+                    })
+                }),
```
Comment (richard-jones) [inline portality/static/js/edges/admin.background_jobs.edge.js:202]: Just one thing to add.  In the `SelectedFilters` component, the `queue_type.exact` fields needs to be mapped to a suitable human readable text.  Probably just "Queue Type" as we have here in the facet

Comment (richard-jones) [reply portality/static/js/edges/admin.background_jobs.edge.js:202]: You can see the effect by selecting a value from the Queue type facet 
Screenshot from 2022-11-05 09-50-38

File: portality/tasks/ingestarticles.py (hunk 1 above)
Comment (philipkcl) [reply portality/tasks/ingestarticles.py:570]: yes, it should be remove.

**Comment by philipkcl** [comment]
updated for review

**Review by amdomanska — APPROVED** [review]

**Review by richard-jones — APPROVED** [review]

**Pull request merged**